manager.go

```diff
@@ -92,12 +89,27 @@ type TaskManager struct {
 	min           int
 	workerTimeout time.Duration
 	stopped       atomic.Bool
+
+	logger Logger
 }
 
+type noLogger struct{}
+
+func (n noLogger) Infof(format string, args ...interface{}) {}
```

Comment:
If you dont use the type in the function you can ignore it

```suggestion
func (noLogger) Infof(format string, args ...interface{}) {}
```